simplecpp.cpp

```diff
@@ -545,31 +555,34 @@ void simplecpp::TokenList::readfile(std::istream &istr, const std::string &filen
 
         // string / char literal
         else if (ch == '\"' || ch == '\'') {
-            // C++11 raw string literal
-            if (ch == '\"' && cback() && cback()->name && isRawStringId(cback()->str())) {
+            if (cback() && cback()->name && !std::isspace(prevChar(istr, bom)) && (isStringLiteralPrefix(cback()->str()))) {
                 std::string delim;
-                ch = readChar(istr,bom);
-                while (istr.good() && ch != '(' && ch != '\n') {
-                    delim += ch;
+                currentToken = ch;
+                bool hasR = *cback()->str().rbegin() == 'R';
+                std::string prefix = cback()->str();
+                if (hasR) {
+                    prefix.resize(prefix.size() - 1);
+                    delim = ")";
                     ch = readChar(istr,bom);
+                    while (istr.good() && ch != '(' && ch != '\n') {
+                        delim += ch;
+                        ch = readChar(istr,bom);
+                    }
+                    if (!istr.good() || ch == '\n')
+                        // TODO report
+                        return;
                 }
-                if (!istr.good() || ch == '\n')
-                    // TODO report
-                    return;
-                currentToken = '\"';
-                const std::string endOfRawString(')' + delim + '\"');
-                while (istr.good() && !endsWith(currentToken, endOfRawString))
+                const std::string endOfRawString(delim + currentToken);
+                while (istr.good() && !(endsWith(currentToken, endOfRawString) && currentToken.size() > 1))
                     currentToken += readChar(istr,bom);
                 if (!endsWith(currentToken, endOfRawString))
                     // TODO report
                     return;
                 currentToken.erase(currentToken.size() - endOfRawString.size(), endOfRawString.size() - 1U);
-                if (cback()->op == 'R')
-                    back()->setstr(escapeString(currentToken));
-                else {
-                    back()->setstr(cback()->str().substr(0, cback()->str().size() - 1));
-                    push_back(new Token(currentToken, location)); // push string without newlines
-                }
+                if (hasR)
+                    currentToken = escapeString(currentToken);
+                currentToken.insert(0, prefix);
+                back()->setstr(currentToken);
                 location.adjust(currentToken);
                 if (currentToken.find_first_of("\r\n") == std::string::npos)
                     location.col += 2 + 2 * delim.size();
```

Comment:
Hm, this location might no longer be correct, I'll take a look.

Reply:
Also,
```C++
#define A L"a\
b"
A;
```
is broken.

Reply:
ok thanks for looking into this...